### PR TITLE
fix(release_health): Do not compare data beyond 90d [INGEST-784]

### PR DESCRIPTION
**Pull request by untitaker**
while investigating failure of get_release_sessions_time_bounds, I
noticed that ALL failures come from the fact that sessions queries are
truncated at 90 days by some snuba/query component. I assume this
affects all service methods to some degree, so let's just cap the
results.

### CODE REVIEW

File: src/sentry/release_health/duplex.py [hunk 1]
```diff
@@ -446,7 +446,16 @@ def __init__(
     ):
         self.sessions = SessionsReleaseHealthBackend()
         self.metrics = MetricsReleaseHealthBackend()
-        self.metrics_start = metrics_start
+        self.metrics_start = min(
```
Comment (jjbayer) [inline src/sentry/release_health/duplex.py:449]:
```suggestion
        self.metrics_start = max(
```
We want to be able to set a more recent metrics_start, but never earlier than 90 days, right?

Comment (untitaker) [reply src/sentry/release_health/duplex.py]: yup thanks

File: src/sentry/release_health/duplex.py [hunk 2]
```diff
@@ -446,7 +446,16 @@ def __init__(
...
-        self.metrics_start = metrics_start
+        self.metrics_start = max(
+            metrics_start,
+            # The sessions backend never returns data beyond 90 days, so any
+            # query beyond 90 days will return truncated results.
+            # We assume that the release health duplex backend is sufficiently
+            # often reinstantiated, at least once per day, not only due to
+            # deploys but also because uwsgi/celery are routinely restarting
+            # processes
+            datetime.now(timezone.utc) - timedelta(days=89),
```
Comment (ahmedetefy) [inline src/sentry/release_health/duplex.py:457]: Why 89 not 90?

**Review by ahmedetefy — APPROVED** [review]

File: src/sentry/release_health/duplex.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/release_health/duplex.py:457]: i didn't want to think about off-by-one errors too much and wanted to err on the side of rather not comparing so we don't have to deal with as many potential false-positives. there's not a lot of mismatches that don't already appear at the 30d range, i assume

**Pull request merged**